src/WakuRlnV2.sol

```diff
@@ -112,14 +149,10 @@ contract WakuRlnV2 {
         idCommitmentIndex += 1;
     }
 
-    function isValidCommitment(uint256 idCommitment) public pure returns (bool) {
-        return idCommitment != 0 && idCommitment < Q;
-    }
-
-    function indexToCommitment(uint32 index) public view returns (uint256) {
-        return imtData.elements[LazyIMT.indexForElement(0, index)];
-    }
-
+    /// @notice Returns the commitments of a range of members
```

Comment:
now that i think about it. having `getCommitments` was meant to be a function to better synchronize the contract, since you no longer have to scan everything looking for events.

but, here we just return the commitment, not the user message limit. so afaik a client wont be able to use this to reconstruct the tree. ofc you can complement if with `idCommitmentToMetadata` but that would involve a lot of calls.

would it make sense to have `getCommitments` return a list of both commitments and usermessagelimit so that a client can easier reconstruct the tree?

Reply:
getCommitments returns the rate commitments which are sufficient to reconstruct the tree.

Reply:
ah yeah my bad. forgot we are already storing the hashed commit/mlimit (aka rate commitment).